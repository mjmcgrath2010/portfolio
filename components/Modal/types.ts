import { ReactElement } from "react";
interface ModalProps {
  children: any;
  buttonText: string | ReactElement;
  size: "sm" | "md" | "lg" | "fullWidth";
}

export interface HeaderProps {
  title: string | ReactElement;
}

export default ModalProps;
