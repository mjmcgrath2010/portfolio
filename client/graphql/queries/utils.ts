/*
 * Example usage:
 *   export const getStaticProps = () => mapQueriesToProps([{ posts: getPosts }]);
 */
type QueryProps = {
  props: Object;
};

export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc: QueryProps, queryObj: Function) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();
          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            props: {
              error: "Oops, something went wrong",
            },
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
