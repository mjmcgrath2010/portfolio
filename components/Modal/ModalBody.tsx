import React from "react";
import { ModalBodyProps } from "./types";
import styled from "styled-components";

const Container = styled.div`
  padding: ${({ theme }) => theme.spacing.md};
`;

const ModalBody = ({ children = "Modal Body" }: ModalBodyProps) => {
  return <Container>{children}</Container>;
};

export default ModalBody;
