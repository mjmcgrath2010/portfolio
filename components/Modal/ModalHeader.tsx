import React from "react";

import Typography from "../Typography";
import { HeaderProps } from "./types";

const ModalHeader = ({ title = "Modal Header" }: HeaderProps) => {
  return <Typography>{title}</Typography>;
};

export default ModalHeader;
