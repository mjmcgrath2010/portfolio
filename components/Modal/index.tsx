import React from "react";
import styled from "styled-components";
import * as Dialog from "@radix-ui/react-dialog";

import ModalProps from "./types";

const Container = styled(Dialog.Root)``;

const Trigger = styled(Dialog.Trigger)`
  ${({ theme }) => theme.variants.button.primary}
`;

const ModalContent = styled(Dialog.Content)`
  width: 300px;
  position: fixed;
  top: 20%;
  left: calc(50% - 150px);
  justify-content: center;
`;

const OverlayContainer = styled(Dialog.Portal)``;

const Modal = (props: ModalProps) => (
  <Container>
    <Trigger />
    <OverlayContainer>
      <Dialog.Overlay />
      <ModalContent>
        <Dialog.Title title="Hello" />
        <Dialog.Description title="Hello" />
        <Dialog.Close />
        Hello
      </ModalContent>
    </OverlayContainer>
  </Container>
);

Modal.defaultProps = {};

export default Modal;
