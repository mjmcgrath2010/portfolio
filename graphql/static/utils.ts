export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc, queryObj) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();

          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            error: e.message,
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
