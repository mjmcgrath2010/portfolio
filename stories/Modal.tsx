
/* eslint-disable import/no-anonymous-default-export */
import React from 'react';
import Modal from '../components/Modal';
import ModalProps from "../components/Modal/types";

export default {
  title: 'Components/Modal',
  component: Modal,
  argTypes: {},
  parameters: {}
};

export const DefaultStory = ({ ...rest }: ModalProps) => <Modal {...rest} />;

DefaultStory.args = {};

DefaultStory.storyName = 'Modal - Default Story';