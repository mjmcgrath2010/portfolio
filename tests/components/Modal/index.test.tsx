import Modal from '@components/Modal';

import { render, screen } from "@test/utils";

describe('<Modal />', () => {
  it("matches the snapshot", () => {
    const view = render(<Modal />);

    expect(view).toMatchSnapshot();
  });
});