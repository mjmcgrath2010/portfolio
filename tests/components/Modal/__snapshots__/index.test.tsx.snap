// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  padding: 16px 16px;
  font-size: 16px;
  outline: none;
  border-radius: 5px;
  border: none;
  cursor: pointer;
  background: rgb(207,109,35);
  color: #fbfbfb;
}

<body>
    <div>
      <button
        aria-controls="radix-:r0:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="c0"
        data-state="closed"
        type="button"
      />
    </div>
  </body>,
  "container": .c0 {
  padding: 16px 16px;
  font-size: 16px;
  outline: none;
  border-radius: 5px;
  border: none;
  cursor: pointer;
  background: rgb(207,109,35);
  color: #fbfbfb;
}

<div>
    <button
      aria-controls="radix-:r0:"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="c0"
      data-state="closed"
      type="button"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
